Predictive analytics with hydra

Ashara-kosi opened the pull request.

Ashara-kosi committed:
Create Predictive-analytics-with deepnote

Ashara-kosi committed:
deepnote-integration

Ashara-kosi committed:
GitBook: No commit message

Ashara-kosi committed:
GitBook: No commit message

Ashara-kosi committed:
GITBOOK-2: change request with no subject merged in GitBook

rundmc5 commented on guides/Predictive-analytics-with deepnote.md:
```diff
@@ -0,0 +1,51 @@
+
+## Setup
+
+- Ensure you have the details of your hydra instance handy.
```
Capitalize Hydra in this case and add a hyperlink to Hydra login

rundmc5 commented on guides/Predictive-analytics-with deepnote.md:
```diff
@@ -0,0 +1,51 @@
+
+### Building the table with Hydra
+
+1. A sample data from [Kaggle](https://www.kaggle.com/datasets/rsadiq/salary) would be used for this guide.Download the csv file.
```
Need to add a space before the second sentence

rundmc5 commented on guides/Predictive-analytics-with deepnote.md:
```diff
@@ -0,0 +1,51 @@
+
+### Creating the Connection
+
+Now we can connect the Postgres Database to Deepnote
```
Can we change this from Postgres to Hydra?

rundmc5 commented on guides/Predictive-analytics-with deepnote.md:
```diff
@@ -0,0 +1,51 @@
+
+Now we can connect the Postgres Database to Deepnote
+
+1. Start by [Signing Up](https://deepnote.com/sign-up) and using the 14 days trial or Logging in if you already have an account.
```
Create your 14 day free trial of Deepnote or sign into your existing Deepnote account.

rundmc5 commented on guides/Predictive-analytics-with deepnote.md:
```diff
@@ -0,0 +1,51 @@
+
+2.  Create a workspace name in this Demo it is called Hydra-Team and choose data source as PostreSQL
+
+Create a new project by clicking the plus sign by the side of projects on the left of the notebook
```
Is it possible to expand these instructions into bullet points with screenshots? Maybe creating another heading that says "Creating your Project"

rundmc5 commented on guides/Predictive-analytics-with deepnote.md:
```diff
@@ -0,0 +1,51 @@
+The dataset is split into dependent and independent variables(X,Y)
+The model is fitted on the dataset and prediction on the salary is made
+Matplotlib is imported to make a visualization of the actual salary and the predicted 
+
```
Would be nice to add a conclusion which includes next steps and a recommendation to reach out for support.

rundmc5 reviewed (commented):
Thanks for the fast work on this Kosi! I think this is heading in the right direction but could use some expansion on the project creation steps. Happy to just on a call to discuss if you have questions on my comments!

Ashara-kosi committed:
GITBOOK-3: change request with no subject merged in GitBook

Ashara-kosi committed:
GITBOOK-{1}: {edit predictive analytics with hydra}

Ashara-kosi committed:
GITBOOK-{1}: {edit predictive analytics with hydra}

wuputah reviewed (changes requested):
The PR should only include one new `.md` file without modifying other files, aside from adding assets to `.gitbook/assets`. You can create a directory in `.gitbook/assets` for organization purposes.

<img width="319" alt="image" src="https://github.com/hydradatabase/docs/assets/3321/7ad18b29-6ce4-44e3-b941-5e9ceb671872">

Ashara-kosi committed:
making edits

Ashara-kosi committed:
making edits

Ashara-kosi committed:
making edits

Ashara-kosi committed:
making edits

Ashara-kosi commented:
I have made the edits that you asked for. The issue before was that my documentation was written in a README.md file named by default on gitbook. I have deleted that file and renamed it in the Predictive-analytics-with-deepnote.md file. Also, I moved the images into the public gitbook/assets folder that was used for other guides.

Please let me know if there are any other issues with the pr.

wuputah committed:
restore README

wuputah committed:
restore guides/README

wuputah committed:
rename file to lowercase

wuputah committed:
remove duplicate files

wuputah committed:
formatting edits

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,116 @@
+  wide range of data frameworks.
+---
+
+# Predictive analytics with Hydra using Deepnote
```
Can we make the main title for this "Introduction"?

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,116 @@
+
+Now we can connect the Postgres Database to Deepnote
+
+1. Start by [Signing Up](https://deepnote.com/sign-up) or Logging in if you already have an account.
```
Can we de-capitalize "signing up" and "logging in"? Thanks

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,116 @@
+Now we can connect the Postgres Database to Deepnote
+
+1. Start by [Signing Up](https://deepnote.com/sign-up) or Logging in if you already have an account.
+2. Create a workspace name, in this Demo, it is called Hydra-Team, and choose the data source as PostgreSQL as shown in the image below.
```
Edit to this sentence: Create a workspace name, which will be "Hydra-Team" in this demo. Next, choose the data source as PostgreSQL as shown in the image below.

rundmc5 commented on .gitbook/assets/.predictive-analytics/Integrating postgres.png:
The screenshots are incredibly helpful! Is it possible to find a more polished arrow in your program? Otherwise another, more standardized way to highlight.

rundmc5 commented on .gitbook/assets/.predictive-analytics/create project.png:
As mentioned previously, could we clean up the arrows with a more standard graphic?

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,116 @@
+
+<figure><img src="../../.gitbook/assets/.predictive-analytics/create project.png" alt=""><figcaption><p>Create a new project</p></figcaption></figure>
+
+4. Make an integration to your Hydra data warehouse by integrating PostgreSQL.Retrieve your database credentials from your Hydra dashboard. You will need Hostname, User, Password, and Database.
```
Edits to the format: Make an integration to your Hydra data warehouse by integrating PostgreSQL.  Retrieve your database credentials from your Hydra dashboard. You will need Hostname, User, Password, and Database.

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,116 @@
+
+<figure><img src="../../.gitbook/assets/.predictive-analytics/connecting postgres.png" alt=""><figcaption><p>Inputing your Hydra credentials</p></figcaption></figure>
+
+5. Once that is connected you can be able to view the tables in your data warehouse in Deepnote and run SQL queries on the salary table that has been imported earlier as shown below using the SELECT statement on the salary table
```
Once that is connected, you can view the tables from your data warehouse in Deepnote and run SQL queries on the salary table that has been imported earlier. The following demonstrates how to use the SELECT statement on the salary table 

This section seems to drop off a bit. Is there an additional action or detail to be added here?

Ashara-kosi replied:
I do not think so. I have edited it to 'After establishing the connection, you can view the imported salary table within Deepnote.
Using the SELECT statement, you can run SQL queries on the salary table, enabling you to retrieve and analyze specific data from the table.
'..hopefully, this is more explanatory?

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,116 @@
+
+## Making the predictions
+
+Now that the Database is connected we can use Python and SQL scripts to access the data. We can now predict the Salary data using the Linear Regression model.&#x20;
```
Might be nice to add a detail about the image below. Is this the linear regression model specifically?

Ashara-kosi replied:
Yes, it is..edited to 'Now that the Database is connected we can use Python and SQL scripts to access the data. Using the Linear Regression model, we can predict the salary data.
The necessary models and libraries are being imported. The linear regression model and matplotlib for our visualization are imported as shown below

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,116 @@
+
+<figure><img src="../../.gitbook/assets/.predictive-analytics/Output.png" alt="" width="464"><figcaption><p>Visuslisation</p></figcaption></figure>
+
+You have now made your first prediction with Hydra using Deepnote!
```
Can we add what could be some next actions now that we have this? You first prediction is made and can help you xyz.

rundmc5 reviewed (commented):
Some small formatting edits but looking so much better! Thanks for the continued effort here, Kosi!

Ashara-kosi committed:
making edits in line with the review

Ashara-kosi committed:
rewriting the statements on the docs

Ashara-kosi committed:
formatting edits

Ashara-kosi committed:
correcting the images

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,119 @@
+
+# Introduction
+
+The focus of this guide is to display the ease of integrating Hydra into [Deepnote](https://deepnote.com/docs) to perform analytics and predictions using Python scripts.
```
The focus of this guide is to display the ease of integrating Hydra into [Deepnote](https://deepnote.com/docs) to perform analytics using Python scripts which can be used to anticipate customer behavior, improve operational efficiency and enhance your business decisions.

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,119 @@
+
+The focus of this guide is to display the ease of integrating Hydra into [Deepnote](https://deepnote.com/docs) to perform analytics and predictions using Python scripts.
+
+A simple salary dataset was chosen with two columns 'Years of experience' and 'Salary' to make predictions.
```
To cater to a broad audience, the data and KPIs were chosen to be as agnostic as possible. We will leverage a simple salary dataset on Kaggle (add link) with two columns:

- 'Years of experience'
- 'Salary'

Your datasets will likely differ in scope and complexity, but this will hopefully serve as a general tutorial on how to leverage your data with Hydra & Deepnote for predictive analytics.

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,119 @@
+
+## Setup
+
+* Ensure you have the details of your Hydra instance handy by creating a data warehouse in Hydra as shown in the [Setup guide](https://docs.hydra.so/getting-started/setup-guide).
```
Ensure you have your Hydra login details handy or follow the [Setup guide](https://docs.hydra.so/getting-started/setup-guide) to create your initial Hydra data warehouse.

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,119 @@
+
+Now we can connect the Postgres Database to Deepnote
+
+1. Start by [signing Up](https://deepnote.com/sign-up) or logging in if you already have an account.
```
Now we can connect Deepnote to our sample data in Hydra.

1. Start by [signing up for Deepnote](https://deepnote.com/sign-up) if you haven't done so already.

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,119 @@
+Now we can connect the Postgres Database to Deepnote
+
+1. Start by [signing Up](https://deepnote.com/sign-up) or logging in if you already have an account.
+2. Create a workspace name, which will be "Hydra-Team" in this demo. Next, choose the data source as PostgreSQL as shown in the image below.
```
2. Within your Deepnote account, create a workspace name. In the case of this tutorial, the workspace name will be "Hydra-Team". 

3. Next, choose 'PostgreSQL' as the data source as shown in the image below.

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,119 @@
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/datasource.png" alt=""><figcaption><p>Selecting PostreSQL</p></figcaption></figure>
+
+3. Create a new project by clicking on projects at the left panel of the screen
```
4. Create a new project by clicking on '+' under 'Projects' within the left panel of the Deepnote console.

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,119 @@
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/create project.png" alt=""><figcaption><p>Create a new project</p></figcaption></figure>
+
+4.  Make an integration to your Hydra data warehouse by integrating PostgreSQL. Retrieve your database credentials from your Hydra dashboard. You will need Hostname, User, Password, and Database.
```
5. Within your new project, navigate to 'Integrations' on the right panel and choose 'PostgreSQL' (followed by screenshot and space)
6. At the 'Connect to PostgresSQL' popup, add your Hydra database credentials including Hostname, User, Password, and Database (followed by screenshot and space)

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,119 @@
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/connecting postgres.png" alt=""><figcaption><p>Inputing your Hydra credentials</p></figcaption></figure>
+
+5. After establishing the connection, you can view the imported salary table within Deepnote.
```
7. After establishing the connection, you can view the imported salary table within Deepnote.

Using the SELECT statement, you are able to run SQL queries on the salary table which enables you to retrieve and analyze specific data from the table.

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,119 @@
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/Hydra dashboard.png" alt=""><figcaption><p>Deepnote dashboard after connection</p></figcaption></figure>
+
+6. To access the table using Python scripts, an engine has to be created with the data warehouse credentials
```
Can we add some details about why this will be necessary for the predictions?

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,119 @@
+
+## Making the predictions
+
+Now that the Database is connected we can use Python and SQL scripts to access the data. Using the Linear Regression model, we can predict the salary data.
```
Now that we have connected Deepnote to Hydra, we can use Python and SQL scripts to access the data. We will use the Linear Regression model to create predictions leveraging the salary data.

Can you add some details here regarding the Linear Regression Model?

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,119 @@
+## Making the predictions
+
+Now that the Database is connected we can use Python and SQL scripts to access the data. Using the Linear Regression model, we can predict the salary data.
+The necessary models and libraries are being imported.The linear regression model and matplotlib for our visualization are imported as shown below&#x20;
```
Add space between sentences

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
````diff
@@ -0,0 +1,119 @@
+from sklearn.linear_model import LinearRegression
+```
+
+The dataset is split into dependent and independent variables(X, Y),the model is trained on the dataset, and prediction on the salary table  is made.
````
Can you expand a bit here to give more details on this approach?

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
````diff
@@ -0,0 +1,119 @@
+Y_pred = model.predict(X)  # make predictions
+```
+
+Visualizations are made on the actual salary and the predicted to view the difference.
````
Visualizations are created to show the difference between the actual salary versus the predicted salary.

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,119 @@
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/Output.png" alt="" width="464"><figcaption><p>Visualization</p></figcaption></figure>
+
+Your first prediction is made and you can determine the range of a person's salary from years of experience.
```
Your first prediction is complete! You can determine the range of a person's salary based on years of experience.

Section - Conclusion
At this point, we have performed a basic predictive analytics project by integrating the Hydra database with Deepnote. This basic guide can be leveraged to experiment with your own business metrics, but don't hesitate to reach out for more advance support through the Hydra #experts channel in Discord.

rundmc5 reviewed (changes requested):
Made some recommended edits and asks for expanded insight to ensure the users feel empowered to perform this integration on their own. We are at the finish line on this one! Thanks Kosi!

Ashara-kosi committed:
Adding more details for clarity

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,135 @@
+
+# Introduction
+
+The focus of this guide is to display the ease of integrating Hydra into [Deepnote](https://deepnote.com/docs) to perform analytics using Python scripts which can be used to anticipate customer behavior, improve operational efficiency and enhance your business decisions.
```
One small edit - can we say "to perform predictive analytics"? I think it is important to call out that this is a data science function vs traditional analtyics

rundmc5 commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,135 @@
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/Output.png" alt="" width="464"><figcaption><p>Visualization</p></figcaption></figure>
+
+Your first prediction is complete! You can determine the range of a person's salary based on years of experience.
```
One more small edit - You can predict the range of a person's salary based on years of experience.

rundmc5 reviewed (changes requested):
Looks good Kosi! Just two small edits. Thanks for all the changes and additional context. Passing this onto JD for final sign off.

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,135 @@
+2. Within your Deepnote account, create a workspace name. In the case of this tutorial, the workspace name will be "Hydra-Team".
+3. Next, choose 'PostgreSQL' as the data source as shown in the image below.
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/datasource.png" alt=""><figcaption></figcaption></figure>
```
```suggestion
![](../.gitbook/assets/predictive-analytics/datasource.png)
```

Please convert images to markdown syntax unless you are also providing a caption or there's some other reason to use this markup. Alt text would also be appreciated for accessibility, i.e.:

```suggestion
![alt text goes here](../.gitbook/assets/predictive-analytics/datasource.png)
```

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,135 @@
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/Hydra dashboard.png" alt=""><figcaption></figcaption></figure>
+
+8. To access the table using Python scripts because predictions can not be done using SQL, an engine has to be created with the data warehouse credentials
```
```suggestion
8. To access the table using Python scripts because predictions can not be done using SQL, an engine has to be created with the data warehouse credentials:
```

However, this is confusing, I don't understand why I am creating this file or where to put it.

Ashara-kosi replied:
A new file is not created, and the connection is done so we can use Python codes(scripts) to access the database on Deepnote instead of SQL, do I need to change anything here?

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,135 @@
+engine = create_engine('postgresql://user:pass@server:5432/database')
+#table_name = Salary
+#database = public
+data = pd.read_sql_table('table_name',con= engine, schema= 'database')
```
this needs to be cleaned up - either remove the commented-out variables or have the commented-out variables be at most two (`credential` and `table_name`) and then used in the code.

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,135 @@
+The goal of linear regression model is to find a mathematical relationship or equation that best fits the data and helps us predict salaries based on the other factors.
+
+The necessary models and libraries are being imported.
+The linear regression model and matplotlib for our visualization are imported as shown below&#x20;
```
```suggestion
The linear regression model and matplotlib for our visualization are imported as shown below:
```

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,135 @@
+
+The goal of linear regression model is to find a mathematical relationship or equation that best fits the data and helps us predict salaries based on the other factors.
+
+The necessary models and libraries are being imported.
```
this sentence doesn't make sense, possibly could be removed?

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,135 @@
+
+## Making the predictions
+
+Now that we have connected Deepnote to Hydra, we can use Python and SQL scripts to access the data. We will use the Linear Regression model to create predictions leveraging the salary data.
+
+The goal of linear regression model is to find a mathematical relationship or equation that best fits the data and helps us predict salaries based on the other factors.
```
suggest combining these into a single paragraph.

Ashara-kosi replied:
You suggested removing the second paragraph in the earlier comment, do I need to go ahead to combine it or remove it?

wuputah commented on guides/predictive-analytics-with-deepnote.md:
````diff
@@ -0,0 +1,135 @@
+from sklearn.linear_model import LinearRegression
+```
+
+The dataset is divided into dependent and independent variables(X, Y), in order for the linear regression model to understand the relationship between the varibles and make predictions.We train the linear regression model using the dataset. During the training process, the model learns from the data by identifying the patterns and relationships between the independent variables (years of experien ...
````
```suggestion
The dataset is divided into dependent (_y_) and independent (_x_) variables in order for the linear regression model to understand the relationship between the variables and make predictions. We train the linear regression model using the dataset. During the training process, the model learns from the data by identifying the patterns and relationships between the independent variables (years of experience) and the dependent variable (the salary).
```

wuputah commented on guides/predictive-analytics-with-deepnote.md:
````diff
@@ -0,0 +1,135 @@
+print(X)
+Y = data.iloc[:,1].values.reshape(-1,1)
+print(Y)
+```
+
+```python
````
probably can combine into a single code block?

wuputah commented on guides/predictive-analytics-with-deepnote.md:
````diff
@@ -0,0 +1,135 @@
+Y_pred = model.predict(X)  # make predictions
+```
+
+Visualizations are created to show the difference between the actual salary versus the predicted salary.
````
```suggestion
Now you can create visualizations to show the difference between the actual salary versus the predicted salary:
```

wuputah commented on .gitbook/assets/.predictive-analytics/Hydra dashboard.png:
filename doesn't make sense to me

rundmc5 replied:
@Ashara-kosi can we update this to Predictive Analytics with Deepnote?

Ashara-kosi committed:
Edits on review

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,124 @@
+
+To cater to a broad audience, the data and KPIs were chosen to be as agnostic as possible. We will leverage a simple salary dataset on [Kaggle](https://www.kaggle.com/datasets/rsadiq/salary) with two columns:
+
+  * 'Years of experience'
+  * 'Salary'
```
```suggestion
  * Years of experience
  * Salary
```

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,124 @@
+
+## Setup
+
+* Ensure you have your Hydra login details handy or follow the [Setup guide](https://docs.hydra.so/getting-started/setup-guide) to create your initial Hydra data warehouse.
```
```suggestion
* Ensure you have your Hydra login details handy or follow the [setup guide](https://docs.hydra.so/getting-started/setup-guide) to create your initial Hydra data warehouse.
```

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,124 @@
+
+* Ensure you have your Hydra login details handy or follow the [Setup guide](https://docs.hydra.so/getting-started/setup-guide) to create your initial Hydra data warehouse.
+
+* Sign up for your free instance at [Deepnote](https://deepnote.com/sign-up)
```
```suggestion
* [Sign up for a free instance at Deepnote](https://deepnote.com/sign-up).
```

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,124 @@
+
+### Building the table with Hydra
+
+1. Sample data from [Kaggle](https://www.kaggle.com/datasets/rsadiq/salary) would be used for this guide. Download the CSV file.
```
```suggestion
1. Sample data from Kaggle will be used for this guide. [Download the CSV file](https://www.kaggle.com/datasets/rsadiq/salary) from their website.
```

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,124 @@
+### Building the table with Hydra
+
+1. Sample data from [Kaggle](https://www.kaggle.com/datasets/rsadiq/salary) would be used for this guide. Download the CSV file.
+2. Connect to your Hydra database and import the CSV file to the database using the steps for importing a CSV file outlined in the [Documentation](https://docs.hydra.so/centralize-data/load/from-local-csv-file)
```
```suggestion
2. Connect to your Hydra database and import the CSV file to the database. To do so, [follow the documentation for importing a CSV file](https://docs.hydra.so/centralize-data/load/from-local-csv-file).
```

You might want to suggest a schema (`CREATE TABLE` statement) here.

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,124 @@
+
+Now we can connect Deepnote to our sample data in Hydra.
+
+1. Start by [signing up](https://deepnote.com/sign-up) for Deepnote if you haven't done so already.
```
```suggestion
1. Start by [signing up for Deepnote](https://deepnote.com/sign-up) if you haven't done so already.
```

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,124 @@
+
+1. Start by [signing up](https://deepnote.com/sign-up) for Deepnote if you haven't done so already.
+2. Within your Deepnote account, create a workspace name. In the case of this tutorial, the workspace name will be "Hydra-Team".
+3. Next, choose 'PostgreSQL' as the data source as shown in the image below.
```
```suggestion
3. Next, choose "PostgreSQL" as the data source as shown in the image below.
```

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,124 @@
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/datasource.png" alt=""><figcaption><p>Choosing Datasource</p></figcaption></figure>
+
+4. Create a new project by clicking on '+' under 'Projects' within the left panel of the Deepnote console.
```
```suggestion
4. Create a new project by clicking on "+" under "Projects" within the left panel of the Deepnote console.
```

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,124 @@
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/create project.png" alt=""><figcaption><p>Creating project</p></figcaption></figure>
+
+5. Within your new project, navigate to 'Integrations' on the right panel and choose 'PostgreSQL'.
```
```suggestion
5. Within your new project, navigate to "Integrations" on the right panel and choose "PostgreSQL".
```

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,124 @@
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/Integrating postgres.png" alt=""><figcaption></figcaption></figure>
+
+6. At the 'Connect to PostgresSQL' popup, add your Hydra database credentials including Hostname, User, Password, and Database.
```
```suggestion
6. In the "Connect to PostgresSQL"' pop-up, add your Hydra database credentials including hostname, user, password, and database. You can obtain these credentials from the [Hydra dashboard](https://dashboard.hydra.so/).
```

wuputah committed:
Update guides/predictive-analytics-with-deepnote.md

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,124 @@
+
+7. After establishing the connection, you can view the imported salary table within Deepnote.
+
+Using the SELECT statement, you are able to run SQL queries on the salary table which enables you to retrieve and analyze specific data from the table.
```
```suggestion
Using a SELECT statement, you can run SQL queries on the salary table which enables you to retrieve and analyze specific data from the table. Let's run a simple query to make sure everything is working like we expect:
```

wuputah commented on guides/predictive-analytics-with-deepnote.md:
````diff
@@ -0,0 +1,124 @@
+```python
+import pandas as pd
+from sqlalchemy import create_engine
+#user = username
+#pass = password
+#server = hostname
+#database = database-name
````
```suggestion
```

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,124 @@
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/deepnote dashboard.png" alt=""><figcaption><p>Deepnote dashboard showing the salary table</p></figcaption></figure>
+
+8. To access the table using Python scripts because predictions can not be done using SQL, an engine has to be created with the data warehouse credentials:
```
```suggestion
8. To access the table using Python scripts because predictions can not be done using SQL, an engine has to be created with the data warehouse credentials. Replace the credential, table name, and schema as appropriate in the example script below:
```

wuputah commented on guides/predictive-analytics-with-deepnote.md:
````diff
@@ -0,0 +1,124 @@
+data = pd.read_sql_table('Salary',con= engine, schema= 'public')
+```
+
+**Congratulations!** You've successfully connected Hydra with Deepnote! What next?
````
```suggestion
Congratulations! You've successfully connected Hydra with Deepnote! What's next?
```

wuputah commented on guides/predictive-analytics-with-deepnote.md:
```diff
@@ -0,0 +1,124 @@
+Your first prediction is complete! You can predict the range of a person's salary based on years of experience.
+
+## Conclusion
+At this point, we have performed a basic predictive analytics project by integrating the Hydra database with Deepnote. This basic guide can be leveraged to experiment with your own business metrics, but don't hesitate to reach out for more advance support through the Hydra #experts channel in Discord.
```
```suggestion
At this point, we have performed a basic predictive analytics project by integrating the Hydra database with Deepnote. This basic guide can be leveraged to experiment with your own business metrics, but don't hesitate to reach out for more advance support through the [Hydra #experts channel in Discord](https://discord.com/invite/SQrwnAxtDw).
```

Ashara-kosi committed:
Update guides/predictive-analytics-with-deepnote.md

Co-authored-by: Jonathan Dance (JD) <wuputah@users.noreply.github.com>

Ashara-kosi committed:
Apply suggestions from code review

Co-authored-by: Jonathan Dance (JD) <wuputah@users.noreply.github.com>
